src/renderer/kube-object.store.ts

```diff
@@ -109,9 +109,11 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
         return api.list({}, this.query);
       }
 
-      const isLoadingAll = this.context.allNamespaces.every(ns => namespaces.includes(ns));
+      const isLoadingAll = namespaces.includes("kube-system")
+                            && this.context.cluster.accessibleNamespaces.length === 0
+                            && this.context.allNamespaces.every(ns => namespaces.includes(ns));
```

Comment:
This would be a complete solution since it covers the case when the namespace is selected from the context in the config and if there is actually only one namespace then it isn't a performance hit to list from it individually.

```suggestion
      const isLoadingAll = this.context.cluster.availableNamespaces.length > 1
                            && this.context.cluster.accessibleNamespaces.length === 0
                            && this.context.allNamespaces.every(ns => namespaces.includes(ns));
```